pkg/watermark/processor/processor_to_fetch.go

```diff
@@ -48,38 +49,48 @@ func (s status) String() string {
 	return "unknown"
 }
 
-// ProcessorToFetch is the smallest unit of entity (from which we fetch data) that does inorder processing or contains inorder data.
+// ProcessorToFetch is the smallest unit of entity (from which we fetch data) that does inorder processing or contains inorder data. It tracks OT for all the partitions of the incoming buffer.
 type ProcessorToFetch struct {
-	ctx            context.Context
-	entity         ProcessorEntitier
-	status         status
-	offsetTimeline *timeline.OffsetTimeline
-	lock           sync.RWMutex
-	log            *zap.SugaredLogger
+	ctx             context.Context
+	entity          ProcessorEntitier
+	status          status
+	offsetTimelines []*timeline.OffsetTimeline
+	lock            sync.RWMutex
+	log             *zap.SugaredLogger
 }
```

Comment:
might be helpful to have a comment for `offsetTimelines` that it's one per partition?